src/python/bot/tasks/train_gradientfuzz_task.py

```diff
@@ -13,9 +13,251 @@
 # limitations under the License.
 """GradientFuzz training task."""
 
+import glob
+import os
+import shutil
+import sys
+
+from bot.fuzzers.ml.gradientfuzz import constants
+from bot.fuzzers.ml.gradientfuzz import run_constants
+from bot.tasks import ml_train_utils
+from build_management import build_manager
+from google_cloud_utils import storage
 from metrics import logs
+from system import environment
+from system import new_process
+from system import shell
+
+# Model script directory.
+GRADIENTFUZZ_SCRIPTS_DIR = os.path.join(
+    os.path.dirname(os.path.dirname(__file__)), 'fuzzers', 'ml', 'gradientfuzz')
+
+
+def get_script_path(which_script):
+  """
+  Gives executable path of requested script.
+
+  Args:
+    which_script (str): Script from `run_constants.py`.
+
+  Returns:
+    (str): Absolute path to requested script (Python file).
+  """
+  return os.path.join(GRADIENTFUZZ_SCRIPTS_DIR, which_script)
+
+
+def get_corpus_directory(root_dir, fuzzer_name):
+  """Gets corpus directory path based on fuzzer."""
+  return os.path.join(root_dir, run_constants.CORPUS_DIR,
+                      fuzzer_name + run_constants.CORPUS_SUFFIX)
+
+
+def upload_model_to_gcs(model_directory, fuzzer_name):
+  """
+  Upload the entire model directory to GCS bucket.
+
+  Note that metadata stored within the model directory is needed
+  for the generation scripts.
+
+  Args:
+    model_directory (str): models/[architecture]/[run-name].
+    fuzzer_name (str): The fuzzer the model is trained for.
+
+  Returns:
+    True if corpus can be acquired and False otherwise.
+  """
+  # Get GCS model path.
+  gcs_model_directory = ml_train_utils.get_gcs_model_directory(
+      run_constants.GRADIENTFUZZ_DIR)
+  if not gcs_model_directory:
+    logs.log_error('Failed to upload model: cannot get GCS model bucket.')
+    return
+
+  # Zip entire model directory and upload.
+  model_dir_name = os.path.basename(model_directory)
+  zipped_dir = shutil.make_archive(model_dir_name, 'zip', model_directory)
+  gcs_model_path = '{}/{}'.format(gcs_model_directory, zipped_dir)
```

Comment:
nit: use python 3 f-strings.

i.e. `f'{gcs_model_directory}/{zipped_dir}'`

Reply:
Ah. I didn't know those existed! Thank you for pointing it out :) Fixed!